app/assets/javascripts/listings/LotteryModalController.js.coffee

```diff
@@ -5,26 +5,27 @@
 LotteryModalController = (
   $scope,
   $state,
-  ListingService,
+  ListingDataService,
+  ListingLotteryService,
   AnalyticsService
 ) ->
-  $scope.listing = ListingService.listing
-  $scope.application = ListingService.application
-  $scope.lotteryBucketInfo = ListingService.lotteryBucketInfo
-  $scope.lotteryRankingInfo = ListingService.lotteryRankingInfo
-  $scope.favorites = ListingService.favorites
+  $scope.listing = ListingDataService.listing
+  $scope.application = ListingDataService.application
```

Comment:
this is the problem that causes the missing lottery number issue in the lottery results modal when accessed from my-applications page

This should be calling ShortFormDataService.application instead of ListingDataService Application. It was a bug introduced in the controller refactor:
https://github.com/Exygy/sf-dahlia-web/pull/1092/files#diff-d2adb468b27b224eb81d8ab85755aeabL35

I tried pulling in the ShortFormApplicationDataService into this controller but got an error for some reason. I have to head offline now but will keep looking into it tomorrow 
![image](https://user-images.githubusercontent.com/11825994/51807318-41450280-223a-11e9-8214-aa7baaf830a5.png)

Reply:
This one is resolved for both modals where LotteryModalController is used.